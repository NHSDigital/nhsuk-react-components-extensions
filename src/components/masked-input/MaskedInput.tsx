import React, { HTMLProps } from 'react';
// note to self - removing this and just having 'nhsuk-react-components' leads to error
// "JSX element type 'FormGroup' does not have any construct or call signatures"
// import FormGroup from 'nhsuk-react-components/dist/esm/util/FormGroup';
// import { FormElementProps } from 'nhsuk-react-components/dist/esm/util/types/FormTypes';
// import { InputWidth } from 'nhsuk-react-components/dist/esm/util/types/NHSUKTypes';
import FormGroup from 'nhsuk-react-components/dist/lib/util/FormGroup';
import { FormElementProps } from 'nhsuk-react-components/dist/lib/util/types/FormTypes';
import { InputWidth } from 'nhsuk-react-components/dist/lib/util/types/NHSUKTypes';
import InputMask, { ReactInputMask } from 'react-input-mask';

import classNames from 'classnames';

type InputMaskRef =
  | string
  | ((instance: ReactInputMask | null) => void)
  | React.RefObject<ReactInputMask>
  | null;

type MaskedInputProps = HTMLProps<HTMLInputElement> &
FormElementProps & {
  width?: InputWidth;
  mask: string;
  maskChar?: string;
  formatChars?: { [character: string]: string };
  alwaysShowMask?: boolean;
  inputRef?: (instance: HTMLInputElement | null) => any;
  ref?: InputMaskRef;
};

// const myFormGroup = ()=>{
//   return(
//     <FormGroup/>
//   )
// }

// const MaskedInput: React.FC<MaskedInputProps> = (props) => (
//   <FormGroup<MaskedInputProps> inputType="input" {...props}>
//     {({ className, width, error, ref, ...rest }:any) => {
//       return(
//       <InputMask
//         className={classNames(
//           'nhsuk-input',
//           { [`nhsuk-input--width-${width}`]: width },
//           { 'nhsuk-input--error': error },
//           className,
//         )}
//         {...rest}
//       />
//     )}}
//   </FormGroup>
// );

const MaskedInput: React.FC<MaskedInputProps> = props => (
  <FormGroup<MaskedInputProps> inputType="input" {...props}>
    {({ className, width, error, ref, ...rest }) => (
      <InputMask
        className={classNames(
          'nhsuk-input',
          { [`nhsuk-input--width-${width}`]: width },
          { 'nhsuk-input--error': error },
          className,
        )}
        {...rest}
      />
    )}
  </FormGroup>
);

MaskedInput.defaultProps = {
  type: 'text',
};

export default MaskedInput;
