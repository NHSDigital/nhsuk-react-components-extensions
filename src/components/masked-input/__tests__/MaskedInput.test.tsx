import React from 'react';
import { render } from '@testing-library/react'
import MaskedInput from '../MaskedInput';

describe('MaskedInput', () => {
  it('matches snapshot', () => {
    const component = render(
      <MaskedInput mask="999 999 9999" name="maskedInput" id="maskedInput" label="NHS Number" />,
    );
    expect(component).toMatchSnapshot();
    expect(component.container.textContent).toBe('NHS Number');
  });
});
