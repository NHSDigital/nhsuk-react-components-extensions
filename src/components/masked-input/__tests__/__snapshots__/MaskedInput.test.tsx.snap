// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskedInput matches snapshot 1`] = `
<div>
  <div
    class="nhsuk-form-group"
  >
    <label
      class="nhsuk-label"
      for="maskedInput"
      id="maskedInput--label"
    >
      NHS Number
    </label>
    <input
      class="nhsuk-input"
      id="maskedInput"
      name="maskedInput"
      type="text"
    />
  </div>
</div>
`;
